File: src/xianyu_cli/commands/order.py
"""Order management commands."""

from __future__ import annotations

import asyncio

import click

from xianyu_cli.core.api_client import TokenExpiredError
from xianyu_cli.core.session import run_api_call
from xianyu_cli.models.envelope import fail, ok
from xianyu_cli.utils._common import console
from xianyu_cli.utils.credential import load_credential
from xianyu_cli.utils.display import print_orders_table


def _require_login(output: str):
    cred = load_credential()
    if not cred or cred.is_expired():
        fail("未登录或凭证已过期，请先执行 xianyu login").emit(output)
        return None
    return cred


@click.group()
def order() -> None:
    """订单管理"""


@order.command("list")
@click.option(
    "--role",
    type=click.Choice(["all", "buyer", "seller"]),
    default="all",
    help="按角色筛选",
)
@click.option("--page", type=int, default=1, help="页码")
@click.pass_context
def list_orders(ctx: click.Context, role: str, page: int) -> None:
    """查看订单列表

    示例: xianyu order list --role seller
    """
    output = ctx.obj.get("output", "rich")
    cred = _require_login(output)
    if not cred:
        return

    data: dict = {"pageNumber": page, "pageSize": 20}
    if role != "all":
        data["role"] = role

    try:
        result = asyncio.run(
            run_api_call(cred, "mtop.taobao.idle.order.list", data)
        )
    except TokenExpiredError:
        fail("登录凭证已过期，请重新执行 xianyu login").emit(output)
        return
    except Exception as e:
        fail(f"获取订单列表失败: {e}").emit(output)
        return

    orders = result.get("orderList", [])

    if output == "rich":
        if orders:
            parsed = [
                {
                    "id": o.get("orderId", ""),
                    "title": o.get("itemTitle", ""),
                    "amount": o.get("totalFee", ""),
                    "status": o.get("statusText", ""),
                    "role": o.get("role", ""),
                }
                for o in orders
            ]
            print_orders_table(parsed)
        else:
            console.print("[yellow]暂无订单[/yellow]")
    else:
        ok({"orders": orders, "page": page}).emit(output)


@order.command()
@click.argument("order_id")
@click.pass_context
def detail(ctx: click.Context, order_id: str) -> None:
    """查看订单详情

    示例: xianyu order detail 123456789
    """
    output = ctx.obj.get("output", "rich")
    cred = _require_login(output)
    if not cred:
        return

    try:
        result = asyncio.run(
            run_api_call(cred, "mtop.taobao.idle.order.detail", {"orderId": order_id})
        )
    except TokenExpiredError:
        fail("登录凭证已过期，请重新执行 xianyu login").emit(output)
        return
    except Exception as e:
        fail(f"获取订单详情失败: {e}").emit(output)
        return

    if output == "rich":
        from rich.panel import Panel

        lines = [
            f"[bold]订单号: {result.get('orderId', order_id)}[/bold]",
            f"商品: {result.get('itemTitle', '')}",
            f"金额: [green]¥{result.get('totalFee', '?')}[/green]",
            f"状态: [yellow]{result.get('statusText', '')}[/yellow]",
            f"买家: {result.get('buyerNick', '')}",
            f"卖家: {result.get('sellerNick', '')}",
        ]
        console.print(Panel("\n".join(lines), title="订单详情", border_style="blue"))
    else:
        ok(result).emit(output)

File: golang/cmd/order.go
package cmd

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"

	"xianyu-cli/core"
	"xianyu-cli/models"
	"xianyu-cli/utils"
)

var orderCmd = &cobra.Command{
	Use:   "order",
	Short: "订单管理",
}

var orderListCmd = &cobra.Command{
	Use:   "list",
	Short: "查看订单列表",
	Run: func(cmd *cobra.Command, args []string) {
		cred := requireLogin()
		if cred == nil {
			return
		}
		role, _ := cmd.Flags().GetString("role")
		page, _ := cmd.Flags().GetInt("page")

		data := map[string]interface{}{"pageNumber": page, "pageSize": 20}
		if role != "all" {
			data["role"] = role
		}

		result, err := core.RunAPICall(cred, "mtop.taobao.idle.order.list", data)
		if err != nil {
			handleAPIError(err)
			return
		}

		orders, _ := result["orderList"].([]interface{})
		if outputMode == "rich" {
			if len(orders) > 0 {
				var parsed []map[string]interface{}
				for _, o := range orders {
					m, _ := o.(map[string]interface{})
					parsed = append(parsed, map[string]interface{}{
						"id": m["orderId"], "title": m["itemTitle"],
						"amount": m["totalFee"], "status": m["statusText"], "role": m["role"],
					})
				}
				utils.PrintOrdersTable(parsed)
			} else {
				fmt.Fprintln(os.Stderr, utils.Yellow.Sprint("暂无订单"))
			}
		} else {
			models.OK(map[string]interface{}{"orders": orders, "page": page}).Emit(outputMode)
		}
	},
}

var orderDetailCmd = &cobra.Command{
	Use:   "detail [order_id]",
	Short: "查看订单详情",
	Args:  cobra.ExactArgs(1),
	Run: func(cmd *cobra.Command, args []string) {
		cred := requireLogin()
		if cred == nil {
			return
		}
		result, err := core.RunAPICall(cred, "mtop.taobao.idle.order.detail", map[string]interface{}{"orderId": args[0]})
		if err != nil {
			handleAPIError(err)
			return
		}

		if outputMode == "rich" {
			fmt.Fprintln(os.Stderr)
			fmt.Fprintf(os.Stderr, "╭── %s ──╮\n", utils.Bold.Sprint("订单详情"))
			fmt.Fprintf(os.Stderr, "│ 订单号: %s\n", utils.Bold.Sprint(result["orderId"]))
			fmt.Fprintf(os.Stderr, "│ 商品: %v\n", result["itemTitle"])
			fmt.Fprintf(os.Stderr, "│ 金额: %s\n", utils.Green.Sprintf("¥%v", result["totalFee"]))
			fmt.Fprintf(os.Stderr, "│ 状态: %s\n", utils.Yellow.Sprintf("%v", result["statusText"]))
			fmt.Fprintf(os.Stderr, "│ 买家: %v\n", result["buyerNick"])
			fmt.Fprintf(os.Stderr, "│ 卖家: %v\n", result["sellerNick"])
			fmt.Fprintln(os.Stderr, "╰────────────╯")
		} else {
			models.OK(result).Emit(outputMode)
		}
	},
}

func init() {
	orderListCmd.Flags().String("role", "all", "按角色筛选 (all/buyer/seller)")
	orderListCmd.Flags().Int("page", 1, "页码")
	orderCmd.AddCommand(orderListCmd)
	orderCmd.AddCommand(orderDetailCmd)
}
